components/dashboard/src/admin/UserDetail.tsx

```diff
@@ -0,0 +1,238 @@
+import { PageWithSubMenu } from "../components/PageWithSubMenu"
+import { getGitpodService } from "../service/service";
+import { adminMenu } from "./admin-menu"
+import { WorkspaceSearch } from "./WorkspacesSearch";
+
+
+export default function UserDetail(p: { user: User }) {
+    const [activity, setActivity] = useState(false);
+    const [user, setUser] = useState(p.user);
+    const [accountStatement, setAccountStatement] = useState<AccountStatement>();
+    const [viewAccountStatement, setViewAccountStatement] = useState(false);
+    const [isStudent, setIsStudent] = useState<boolean>();
+    const [editFeatureFlags, setEditFeatureFlags] = useState(false);
+    const [editRoles, setEditRoles] = useState(false);
+    const userRef = useRef(user);
+
+    const isProfessionalOpenSource = accountStatement && accountStatement.subscriptions.some(s => s.planId === Plans.FREE_OPEN_SOURCE.chargebeeId)
+
+    useEffect(() => {
+        setUser(p.user);
+        getGitpodService().server.adminGetAccountStatement(p.user.id).then(
+            as =>
+                setAccountStatement(as)
+        ).catch(e => {
+            console.error(e);
+        });
+        getGitpodService().server.adminIsStudent(p.user.id).then(
+            isStud => setIsStudent(isStud)
+        )
+    }, [p.user]);
+
+    const updateUser: UpdateUserFunction = async fun => {
+        setActivity(true);
+        try {
+            setUser(await fun(userRef.current));
+        } finally {
+            setActivity(false);
+        }
+    };
+
+    const toggleBlockUser = async () => {
+        await updateUser(async u => {
+            u.blocked = !u.blocked;
+            await getGitpodService().server.adminBlockUser({
+                blocked: u.blocked,
+                id: u.id
+            });
+            return u;
+        });
+    }
+
+    const deleteUser = async () => {
+        await updateUser(async u => {
+            u.markedDeleted = !u.markedDeleted;
+            await getGitpodService().server.adminDeleteUser(u.id);
+            return u;
+        });
+    }
+
+    const flags = getFlags(user, updateUser);
+    const rop = getRopEntries(user, updateUser);
+
+    return <>
+        <PageWithSubMenu subMenu={adminMenu} title="Users" subtitle="Search and manage all users.">
+            <div className="flex">
+                <div className="flex-1">
+                    <div className="flex"><h3>{user.fullName}</h3>{user.blocked ? <Label text='Blocked' color="red" /> : null} {user.markedDeleted ? <Label text='Deleted' color="red" /> : null}</div>
+                    <p>{user.identities.map(i => i.primaryEmail).filter(e => !!e).join(', ')}</p>
+                </div>
+                <button className="secondary danger ml-3" disabled={activity} onClick={toggleBlockUser}>{user.blocked ? 'Unblock' : 'Block'} User</button>
+                <button className="danger ml-3" disabled={activity} onClick={deleteUser}>Delete User</button>
```

Comment:
**question**: What do you think of reverting the actions here so that _Block User_ is the primary action instead of _Delete User_. Feel free to leave this as is.

Reply:
I switched that because I felt like "Delete User" is the more severe (not revertable) action. Maybe we should not have that action at all?

Reply:
We could hide less prominent actions in a more actions dropdown button as we do for Workspaces. It should be fine to ship it as is and improve later. ➿